dali/pipeline/operators/fused/new_crop_mirror_normalize.cc

```diff
@@ -0,0 +1,235 @@
+    if (in_layout == DALI_NHWC && out_layout == DALI_NCHW) {
+      perm_dims[0] = 2;
+      perm_dims[1] = 0;
+      perm_dims[2] = 1;
+    } else if (in_layout == DALI_NCHW && out_layout == DALI_NHWC) {
+      perm_dims[0] = 1;
+      perm_dims[1] = 2;
+      perm_dims[2] = 0;
+    } else if (in_layout == DALI_NFHWC && out_layout == DALI_NFCHW) {
+      perm_dims[1] = 3;
+      perm_dims[2] = 1;
+      perm_dims[3] = 2;
+    } else if (in_layout == DALI_NFCHW && out_layout == DALI_NFHWC) {
+      perm_dims[1] = 2;
+      perm_dims[2] = 3;
+      perm_dims[3] = 1;
+    } else {
+      DALI_FAIL("layout conversion from " + std::to_string(in_layout) + " to "
+        + std::to_string(out_layout) + " not supported");
+    }
+  }
+
+  return perm_dims;
+}
+
+size_t channels_dim(DALITensorLayout in_layout) {
+  switch (in_layout) {
+    case DALI_NHWC:
+      return 2;
+    case DALI_NCHW:
+      return 0;
+    case DALI_NFHWC:
+      return 3;
+    case DALI_NFCHW:
+      return 1;
+    default:
+      DALI_FAIL("not supported layout: " + std::to_string(in_layout));
+  }
+}
+
+template <typename OutputType, typename InputType>
+void RunHelper(Tensor<CPUBackend> &output,
+               const Tensor<CPUBackend> &input,
+               const std::vector<int64_t> &slice_anchor,
+               const std::vector<int64_t> &slice_shape,
+               bool horizontal_flip,
+               bool pad_output,
+               const std::vector<float> &mean,
+               const std::vector<float> &inv_std_dev) {
+  std::size_t number_of_dims = input.shape().size();
+  auto input_layout = input.GetLayout();
+  auto output_layout = output.GetLayout();
+  VALUE_SWITCH(number_of_dims, Dims, (3, 4), (
+    auto in_view = view<const InputType, Dims>(input);
+
+    kernels::SliceFlipNormalizePermuteArgs<OutputType, Dims> args;
+    auto &anchor = args.anchor;
+    auto &shape = args.shape;
+    for (std::size_t d = 0; d < Dims; d++) {
+      anchor[d] = slice_anchor[d];
+      shape[d] = slice_shape[d];
+    }
+
+    if (pad_output) {
+      args.should_pad = true;
+      args.padded_shape = args.shape;
+      args.padded_shape[channels_dim(input_layout)] = 4;
+    }
+
+    if (horizontal_flip) {
+      args.should_flip = true;
+      auto flip_dim = horizontal_dim_idx(input_layout);
+      for (size_t d = 0; d < Dims; d++) {
+        args.flip[d] = (d == flip_dim);
+      }
+    }
+
+    // Check if permutation is needed
+    if (input_layout != output_layout) {
+      args.should_permute = true;
+      args.permuted_dims = permuted_dims<Dims>(input_layout, output_layout);
+    }
+
+    args.should_normalize =
+         !std::all_of(mean.begin(), mean.end(), [](float x){ return x == 0.0f; })
+      || !std::all_of(inv_std_dev.begin(), inv_std_dev.end(), [](float x){ return x == 1.0f; });
+
+    if (args.should_normalize) {
+      args.mean = mean;
+      args.inv_stddev = inv_std_dev;
+      args.normalization_dim = channels_dim(input_layout);
```

Comment:
```suggestion
      bool uniform_normalization = mean.size() == 1;
      args.normalization_dim = uniform_normalization ? args.normalization_dim_scalar : channels_dim(input_layout);
```
and add a negative constant `normalization_dim_scalar` to Args, so it never matches `d` in the implementation.

Reply:
done in the kernel implementation